Modules/Core/Common/include/itkPointSetBase.h

```diff
@@ -190,6 +192,25 @@ class ITK_TEMPLATE_EXPORT PointSetBase : public DataObject
 
   itkGetConstMacro(BufferedRegion, RegionType);
 
+
+  /** Returns a "deep copy" of this point set, having copied its points and point data. */
+  Pointer
+  MakeDeepCopy() const
+  {
+    const auto clone = this->Clone();
+    clone->Graft(this);
+
+    if (m_PointsContainer)
```

Comment:
Why not put this into a method `DetachPoints`? The could here would then read:
```cpp
clone->DetachPoints();
clone->DetachPointData();
```
Which is clearer.

Reply:
Well observed, @dzenanz I added `DetachPointData()` as a virtual member function of `PointSetBase` is because it has to be implemented by its derived `PointSet` class (as an `override`). `PointSetBase` cannot detach its point data directly, because it is managed by its derived class. On the other hand, `PointSetBase` _is_ able to detach its own _points_, though, so it does not need the virtual/override mechanism.

With this in mind, would you still like it if we would add a _non_-virtual private `DetachPoints()` member function to `PointSetBase`?

Reply:
non-virtual **proteced** DetachPoints() member function to PointSetBase would be my preference

When looking at this code at first, I thought you are copying points twice. Then I realized once is for `points`, the other time for `point data`.

Reply:
I would prefer to only make a member _protected_ when it is actually necessary to be used by a derived class. I find the `private` keyword much clearer, in order to communicate that the member is _for internal use only_. We could initially make `DetachPoints()` `private`, and then maybe later change it to `protected`, but only when it is really needed for derived classes. Would that be OK to you?